exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java

```diff
@@ -70,11 +69,13 @@
  *  </p>
  */
 
-public class SystemOptionManager extends BaseOptionManager implements OptionManager, AutoCloseable {
+public class SystemOptionManager extends BaseOptionManager implements AutoCloseable {
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(SystemOptionManager.class);
 
-  private CaseInsensitiveMap<OptionValidator> VALIDATORS;
-  public void populateValidators() {
+  public static final CaseInsensitiveMap<OptionValidator> DEFAULT_VALIDATORS =
```

Comment:
A compromise is to introduce a two-level table.

* Global table of option metadata. Perhaps evolve the validator to be that metadata. Or, better, define a new `OptionDefinition` class, an attribute of which is the validator.
* Per-Drillbit table of defaults resolved from the config system using the information from the definitions.

If we do this, we should clean up all the messy enums in this system. The current ones confuse everyone. We want:

* Definition: The allowed scope: session-only or both session and system.
* Visibility: public or hidden (this change)
* Data type: (long, double, string, etc.)

For a value:

* Scope: default, session or system.

Sometimes it is easier to fix the original problems than to keep working around an outdated design...

Reply:
I don't completely follow this. Let's talk about this offline next week so that I can understand :). I can post the conclusion of the discussion on the PR for the community then.